vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+calculation of nadir, change in & percentage change from baseline, and the
+analysis flags that could be setup for reporting.  The tumor results data could
+be set up for investigator and/or Independent review facility (IRF)/Blinded
+Independent Central Review (BICR) data.
+
+The source dataset used will depend on each company,  this could be solely the
+TR domain or you may merge `TU` with `TR` to get additional data variables or to
+ensure that you're processing the same lesions as collected.
+
+Individual sum of diameter for each target lesion (from `TR`) is required to
+calculate the sum of all target lesions, this data could be taken directly from
+`TR` or additional parameters could be created in ADTR (or similar) depending on
+the additional processing required (e.g. imputation of dates, re-labeling of
+visits) and your company specifications.
+
+NEED TO UPDATE THIS PART ONCE TEMPLATE CODE IS IN. The majority of the functions
+used here exist from `{admiral}`, except for the `tte_sources` helper object,
+provided as an example from `{admiralonco}`.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless
+otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages. (This is to be
+reviewed/updated once template code is written)
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(dplyr)
+library(admiral.test)
+library(lubridate)
+library(stringr)
+library(admiralonco)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Merge TR with TU and derive new variables](#merge)
+* [Pre-processing of input records](#input)
+* [Derive Parameters (`TLDn`, `TLPn`)](#parameters)
+* [Derive Parameters (TSD) and ANL01FL](#tsd)
+* [Derive Baseline (BASETYPE, ABLFL, BASE)](#base)
+* [Derive NADIR](#nadir)
+* [Derive Change from Baseline (CHG, PCHG)](#chg)
+* [Derive Additional Flag Variables (e.g. CRFL, CRNFL, PDFL, PDFL](#addflag)
+* [Derive Additional Analysis Flags (e.g. ANL02FL, ANL03FL, ANL04FL](#analflag)
+* [Derive Analysis Sequence Number (`ASEQ`)](#aseq)
+* [Add ADSL Variables](#adslvars)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of the tumor results dataset
+should be read into the environment. This will be a company specific process.
+Some of the data frames needed may be `ADSL`, `ADRS` ,`RS`,  `TU` and `TR`.
+
+For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot test
+data)---which are included in `{admiral.test}`---are used
+
+```{r}
+data("admiral_adsl")
+data("admiral_adrs")
+data("admiral_rs")
+data("admiral_tu")
+data("admiral_tr")
+adsl <- admiral_adsl
+adrs <- admiral_adrs
+tu <- admiral_tu
+tr <- admiral_tr
+rs <- admiral_rs
+
+tu <- convert_blanks_to_na(tu) %>%
+  filter(TUEVAL == "INVESTIGATOR")
+tr <- convert_blanks_to_na(tr) %>%
+  filter(TREVAL == "INVESTIGATOR" & TRGRPID == "TARGET" & TRTESTCD == "DIAMETER")
```

Comment:
I agree, and also, are we presuming users are aware of each column and xxEVAL, xxGRPID and xxTESTCDs mean. Or should we add a brief description?

Reply:
Agree, restrict as 'read in' step.  
On top part of the vignette, I mention independent review data - but I can add more details about setting up code twice, unless there is another suggestion?
Also had more comments to the template code.

Reply:
@amitjaingsk for merging SUPPTU /SUPPTR was there any specific example we could mention?

Reply:
@HinalPat I missed this comment about SUPPxx before, the example you can mention like Tumor laterality info like 'right", "left" info sometime is mapped in SUPPTU. so that can be mentioned as in the text. As its very specific to the study/company, we can mention as a text , otherwise its fine.

Reply:
@amitjaingsk I have now left it out without the example